src/cls/rbd/cls_rbd.cc

```diff
@@ -8091,6 +8091,81 @@ int sparsify(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
   return 0;
 }
 
+#define RWLCACHE_MAP_KEY "rwlcache_map"
+
+/**
+ * Report rwlcache daemon-info
+ *
+ * Input:
+ * @param daemon info (cls::rbd::RwlCacheDaemonInfo)
+ *
+ * Output:
+ * @return -EEXIST if daemon_id or rdma_address&rdma_port existed.
+ * @returns 0 on success, negative error code on failure
+ *
+ */
+int rwlcache_daemoninfo(cls_method_context_t hctx, bufferlist *in, bufferlist *out)
+{
+  cls::rbd::RwlCacheDaemonInfo info;
+  cls_rbd_rwlcache_map map;
+
+  try {
+    auto it = in->cbegin();
+    decode(info, it);
+  } catch (const ceph::buffer::error &err) {
+    return -EINVAL;
+  }
+
+  CLS_LOG(20, "new rwlcache daemon id=%lu, rdma_address=%s:%u, total_size=%lu",
+      info.id, info.rdma_address.c_str(), info.rdma_port, info.total_size);
+
+  bufferlist value;
+  int r = cls_cxx_map_get_val(hctx, RWLCACHE_MAP_KEY, &value);
+  if ((r < 0) && (r != -ENOENT)) {
+    CLS_ERR("error reading key %s: %s", RWLCACHE_MAP_KEY, cpp_strerror(r).c_str());
+    return r;
+  }
+
+  if (r >= 0) {
+    try {
+      auto it = value.cbegin();
+      decode(map, it);
+    } catch (ceph::buffer::error &err) {
+      CLS_ERR("decode %s error", RWLCACHE_MAP_KEY);
+      return -EIO;
+    }
+  }
+
+  if (map.daemons.count(info.id)) {
+    CLS_LOG(5, "existed daemon id (%lu)", info.id);
+    return -EEXIST;
+  }
+
+  for (const auto &[id, daemon] : map.daemons) {
+    if ((daemon.rdma_address == info.rdma_address) &&
```

Comment:
Currently we should only one daemon in a machine(at least i don't think of any reason). If so, we only check rdma_address whether equal.

Reply:
There are two cpu sockets on one server, which means we can get at least two pmem dax devices on one server. Then we can deploy two replica daemons on one server.

And only rdma_address is not friendly to debug on one server.